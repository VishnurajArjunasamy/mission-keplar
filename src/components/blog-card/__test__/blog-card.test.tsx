import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import BlogCard from "../blog-card";

const mockStore = configureStore([]);

describe("BlogCard Component", () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore({
      sideBar: {
        isDarkMode: false,
      },
    });
  });

  it("renders BlogCard with light mode styles", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BlogCard
          title="Test Blog"
          details="This is a test blog."
          type="Tech"
          onClick={() => {}}
          active={false}
        />
      </Provider>
    );

    expect(screen.getByText("Test Blog")).toBeInTheDocument();
    expect(screen.getByText("This is a test blog.")).toBeInTheDocument();
    expect(screen.getByText("Tech")).toBeInTheDocument();
    const blogCard = screen.getByText("Test Blog").parentElement;
    expect(blogCard).toHaveClass("blogCard");
    expect(blogCard).toHaveClass("light");
    expect(blogCard).not.toHaveClass("dark");

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders BlogCard with dark mode styles", () => {
    store = mockStore({
      sideBar: {
        isDarkMode: true,
      },
    });

    const { asFragment } = render(
      <Provider store={store}>
        <BlogCard
          title="Test Blog"
          details="This is a test blog."
          type="Tech"
          onClick={() => {}}
          active={false}
        />
      </Provider>
    );

    const blogCard = screen.getByText("Test Blog").parentElement;
    expect(blogCard).toHaveClass("dark");
    expect(blogCard).not.toHaveClass("light");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies the active class when active prop is true", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BlogCard
          title="Test Blog"
          details="This is a test blog."
          type="Tech"
          onClick={() => {}}
          active={true}
        />
      </Provider>
    );

    const blogCard = screen.getByText("Test Blog").parentElement;
    expect(blogCard).toHaveClass("active");

    expect(asFragment()).toMatchSnapshot();
  });

  it("calls the onClick function when clicked", () => {
    const handleClick = jest.fn();

    const { asFragment } = render(
      <Provider store={store}>
        <BlogCard
          title="Test Blog"
          details="This is a test blog."
          type="Tech"
          onClick={handleClick}
          active={false}
        />
      </Provider>
    );

    const blogCard = screen.getByText("Test Blog").parentElement;
    if (blogCard) {
      fireEvent.click(blogCard);
    }

    expect(handleClick).toHaveBeenCalledTimes(1);

    expect(asFragment()).toMatchSnapshot();
  });
});
