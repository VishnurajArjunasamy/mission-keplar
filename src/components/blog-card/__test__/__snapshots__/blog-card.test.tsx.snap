// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard Component applies the active class when active prop is true 1`] = `
<DocumentFragment>
  <div
    class="blogCard light active"
  >
    <h1>
      Test Blog
    </h1>
    <h2>
      Tech
    </h2>
    <p>
      This is a test blog.
    </p>
  </div>
</DocumentFragment>
`;

exports[`BlogCard Component calls the onClick function when clicked 1`] = `
<DocumentFragment>
  <div
    class="blogCard light false"
  >
    <h1>
      Test Blog
    </h1>
    <h2>
      Tech
    </h2>
    <p>
      This is a test blog.
    </p>
  </div>
</DocumentFragment>
`;

exports[`BlogCard Component renders BlogCard with dark mode styles 1`] = `
<DocumentFragment>
  <div
    class="blogCard dark false"
  >
    <h1>
      Test Blog
    </h1>
    <h2>
      Tech
    </h2>
    <p>
      This is a test blog.
    </p>
  </div>
</DocumentFragment>
`;

exports[`BlogCard Component renders BlogCard with light mode styles 1`] = `
<DocumentFragment>
  <div
    class="blogCard light false"
  >
    <h1>
      Test Blog
    </h1>
    <h2>
      Tech
    </h2>
    <p>
      This is a test blog.
    </p>
  </div>
</DocumentFragment>
`;
